scientific_library/tvb/simulator/simulator.py

```diff
@@ -557,7 +559,8 @@ def _configure_history(self, initial_conditions):
                 self.current_step += ic_shape[0] - 1
 
         # Make sure that history values are bounded
-        self.bound_and_clamp(history)
+        for it in range(history.shape[0]):
+            history[it] = self.bound_and_clamp(history[it])
```

Comment:
The bound_and_clamp method doesn't have a return statement so after this loop all history elements will be nan. There is also the test_propagation in history_test.py which is failing after this change. Call only the bound and clamp method without assigning the method's result to the history[it] because the history[it] values will be changed inside of this method.